core/src/avm2/globals/flash/display/graphics.rs

```diff
@@ -1588,17 +1605,17 @@ fn handle_gradient_fill<'gc>(
     activation: &mut Activation<'_, 'gc>,
     obj: &Object<'gc>,
 ) -> Result<FillStyle, Error<'gc>> {
-    let alphas = obj
-        .get_public_property("alphas", activation)?
-        .coerce_to_object(activation)?;
+    let alphas = obj.get_public_property("alphas", activation)?.as_object();
 
     let colors = obj
         .get_public_property("colors", activation)?
-        .coerce_to_object(activation)?;
+        .as_object()
+        .ok_or_else(|| make_error_2007(activation, "colors"))?;
 
     let ratios = obj
         .get_public_property("ratios", activation)?
-        .coerce_to_object(activation)?;
+        .as_object()
+        .expect("Ratios should never be null");
```

Comment:
I didn't check what FP does here (probably either throws or interpolates itself), but I think I can write AS that'll put a null here

Reply:
See previous review